Graphics/ShapeFactory.java

```diff
@@ -0,0 +1,46 @@
+package Graphics;
+
+import java.util.List;
+
+import Graphics.Shape.ShapeType;
+
+/**
+ * @author Niraj Gupta
+ * Class to implement the abstract factory to create the reference to the object of shape type like
+ * Rectangle, Square, Circle, Regular Polygon and Triangle.
+ */
+public class ShapeFactory{
+
+	/**
+	 * Method to create the reference to the object of shape type on the basis of requirement.
+	 * @param shapeType : The type of the shape whose object has to be created.
+	 * @param point : The point at which the shape is placed on the screen.
+	 * @param list : The list containing the basic attributes of the shape. For example the list contain length of side if the 
+	 * 				 shape type is square or radius if the shape type is circle etc.
+	 * @return : The reference to the object of the shape type.
+	 */
+	public static Shape createShape(ShapeType shapeType, Point point, List<Integer> list) {

```

Comment:
How appropriate is it to name a list of integer as list?

Reply:
It seems fine to me. But according to you what can be the appropriate name for the list.